tasks/transform_conformance/tests/legacy-decorators/test/fixtures/oxc/use-define-for-class-fields/output.js

```diff
@@ -0,0 +1,10 @@
+class Cls {
+  constructor() {
+    this.y = 1;
+  }
+}
+babelHelpers.decorate([dce], Cls.prototype, "z", void 0);
+
+class StaticCls { }
+babelHelpers.decorate([dce], StaticCls, "z", void 0);
+StaticCls.y = 1;
```

Comment:
```suggestion
StaticCls.y = 1;
babelHelpers.decorate([dce], StaticCls, "z", void 0);
```
I guess this has to be in the reverse order.
[TS playground](https://www.typescriptlang.org/play/?experimentalDecorators=true&useDefineForClassFields=false#code/MYGwhgzhAEDKAuZ4EtgGEQwN4CgCQEiKw0AHgFzQB2ArgLYBGApgE4Dc+hSq0AntAF5oARg54AAgBNgTTkR4AvDgF8gA)
(probably this can be tackled separately)

Reply:
I am not sure whether it implies runtime, but what is certain is that adjusting the order is not easy and has some cost. I prefer to keep it now.

Reply:
Here's a example that the behavior changes.
```ts
function decorator(target: any, propertyKey: string) {
  console.log('y', target.y)
}

class StaticCls {
	static y = 1;
	@decorator
	static z: string;
}
```
[TS playground](https://www.typescriptlang.org/play/?experimentalDecorators=true&useDefineForClassFields=false#code/GYVwdgxgLglg9mABAEwKYTgJwIZSwCim0wHNUoAuRbMATwBpEAHTOJ1TKWgaVVqoDOUTDDAkAlIgDeAKESIMYAXAA2qAHQq4JfAHJauxkVLl1tcTIC+MmRBXYBAxAGUisCAGEVT2QEghuDAQiLSIALyIAIwA3DK+AAJoGDh4mHEB7ogAXoLCoiSx1kA)

I don't think people would rely on this behavior so I think it's fine to keep it, too.

I noticed that TypeScript doesn't keep the execution order in some cases as well ([example](https://www.typescriptlang.org/play/?useDefineForClassFields=true&experimentalDecorators=true&target=99#code/GYVwdgxgLglg9mABAEwKYTgJwIZSwCim0wHNUoAuRbMATwBpEAHTOJ1TKWgaVVqoDOUTDDAkAlIgDeAKESIMYAXAA2qAHQq4JfAHJauxkVLl1tcTIC+MmRBXYBAxAGUisCAGEVT2QEghuDAQiLSIALyIAIwA3DK+AAJoGDh4mHEB7ogAXoLCoiSx-m5B2QBM4VGx1kA)).

Reply:
I see. I think it should handle decorator initialization like handling computed keys does, but still very complicated because it separated between two plugins. Let's keep it now.